x-pack/plugins/ux/public/components/app/rum_dashboard/charts/visitor_breakdown_chart.tsx

```diff
@@ -5,90 +5,216 @@
+  OS_BREAKDOWN = 'user_agent.os.name',
+  UA_BREAKDOWN = 'user_agent.name',
 }
 
-const theme: PartialTheme = {
-  chartMargins: { top: 0, bottom: 0, left: 0, right: 0 },
-  legend: {
-    verticalWidth: 100,
-  },
-  partition: {
-    linkLabel: { maximumSection: Infinity, maxCount: 0 },
-    outerSizeRatio: 1, // - 0.5 * Math.random(),
-    circlePadding: 4,
-  },
-};
+interface LensAttributes {
+  metric: VisitorBreakdownMetric;
+  uiFilters: UxUIFilters;
+  urlQuery?: string;
+  dataView: string;
+}
+
+type Props = {
+  start: string;
+  end: string;
+  onFilter: (metric: VisitorBreakdownMetric, event: any) => void;
+} & LensAttributes;
+
+export function VisitorBreakdownChart({
+  start,
+  end,
+  onFilter,
+  uiFilters,
+  urlQuery,
+  metric,
+  dataView,
+}: Props) {
+  const kibana = useKibanaServices();
+  const LensEmbeddableComponent = kibana.lens.EmbeddableComponent;
 
-export function VisitorBreakdownChart({ loading, options }: Props) {
-  const [darkMode] = useUiSetting$<boolean>('theme:darkMode');
+  const lensAttributes = useMemo(
+    () =>
+      getVisitorBreakdownLensAttributes({
+        uiFilters,
+        urlQuery,
+        metric,
+        dataView,
+      }),
+    [uiFilters, urlQuery, metric, dataView]
+  );
+
+  const filterHandler = useCallback(
+    (event) => {
+      onFilter(metric, event);
+    },
+    [onFilter, metric]
+  );
 
-  const euiChartTheme = darkMode
-    ? EUI_CHARTS_THEME_DARK
-    : EUI_CHARTS_THEME_LIGHT;
+  if (!LensEmbeddableComponent) {
+    return <EuiText>No lens component</EuiText>;
+  }
 
   return (
-    <ChartWrapper loading={loading} height="245px" maxWidth="430px">
-      <StyleChart>
-        <Chart>
-          <Settings
-            showLegend
-            baseTheme={darkMode ? DARK_THEME : LIGHT_THEME}
-            theme={theme}
-          />
-          <Partition
-            id="spec_1"
-            data={
-              options?.length ? options : [{ count: 1, name: I18LABELS.noData }]
-            }
-            layout={PartitionLayout.sunburst}
-            clockwiseSectors={false}
-            valueAccessor={(d: Datum) => d.count as number}
-            valueGetter="percent"
-            percentFormatter={(d: number) =>
-              `${Math.round((d + Number.EPSILON) * 100) / 100}%`
-            }
-            layers={[
-              {
-                groupByRollup: (d: Datum) => d.name,
-                shape: {
-                  fillColor: (d) =>
-                    euiChartTheme.theme.colors?.vizColors?.[d.sortIndex]!,
-                },
-              },
-            ]}
-          />
-        </Chart>
-      </StyleChart>
-    </ChartWrapper>
+    <LensEmbeddableComponent
+      id={`ux-visitor-breakdown-${metric.replaceAll('.', '-')}`}
+      hidePanelTitles
+      withDefaultActions
+      style={{ minHeight: '250px', height: '100%' }}
+      attributes={lensAttributes}
+      timeRange={{
+        from: start ?? '',
+        to: end ?? '',
+      }}
```

Comment:
i think we shouldn't render if start and end are missing, might end up hitting query without time range.